fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -849,4 +851,34 @@ public void replayModifyTableColocate(TablePropertyInfo info) {
             db.writeUnlock();
         }
     }
+
+    /**
+     * for legacy reason, all the colocate group index cannot be properly removed
+     * we have to add a cleanup function on start when loading image
+     */
+    protected void cleanupInvalidDbOrTable(GlobalStateMgr globalStateMgr) {
+        List<Long> badTableIds = new ArrayList<>();
+        for (Map.Entry<Long, GroupId> entry : table2Group.entrySet()) {
+            long dbId = entry.getValue().dbId;
+            long tableId = entry.getKey();
+            Database database = globalStateMgr.getDbIncludeRecycleBin(dbId);
+            if (database == null) {
```

Comment:
Where to remove the dbId?

Reply:
Normally database is removed by the CatalogRecycleBin and CatalogRecycle will remove the colocate index from memory as well. But for those who have just upgraded from an older version, we should provide a mechanism to remove colocate index that remains in memory because of this bug. So I use this function to do this one-time job when loading an image.